spec/shared/cassette_secrets_helper.rb

```diff
@@ -0,0 +1,32 @@
+module Spec
+  module Shared
+    module CassetteSecretsHelper
+      DEFAULT_VCR_SECRETS_PATH = Pathname.new(Dir.pwd).join("config/secrets.defaults.yml")
+      VCR_SECRETS_PATH         = Pathname.new(Dir.pwd).join("config/secrets.yml")
```

Comment:
I think this might be a little cleaner, just in case Dir.pwd isn't what we think it is?

```suggestion
      if defined?(ENGINE_ROOT)
        DEFAULT_VCR_SECRETS_PATH = ENGINE_ROOT.join("config/secrets.defaults.yml")
        VCR_SECRETS_PATH         = ENGINE_ROOT.join("config/secrets.yml")
      end
```

Since this is at the constant level, that's really all we can do, unless we also want to allow `Rails.root` and allow core to have secrets too.  Otherwise, if we move this into a method instead, and load it on-demand, then we could do something like:

```suggestion
      def default_vcr_secrets_path
        raise "Cassette secrets are only expected to be used from plugins" unless defined?(ENGINE_ROOT)
        
        ENGINE_ROOT.join("config/secrets.defaults.yml")
      end

      def vcr_secrets_path
        raise "Cassette secrets are only expected to be used from plugins" unless defined?(ENGINE_ROOT)
        
        ENGINE_ROOT.join("config/secrets.yml")
      end
```

Reply:
`defined?(ENGINE_ROOT)` basically means "we are running this from a plugin" as opposed to "we are running in core"

Reply:
Thanks, I thought I tried ENGINE_ROOT.  I'll take another look.

Reply:
I prefer the 'extract a method' style as we change it if it needs to be handled in core.

Reply:
oh, I remember the hurdle I hit.  ENGINE_ROOT is not defined.  I could find it but it's the same as what I did:

```
(byebug) Vmdb::Plugins.paths.detect {|_,path| File.expand_path(path) == File.expand_path(Dir.pwd)}.last
"/Users/joerafaniello/Code/manageiq-providers-vmware"
```

Reply:
I can take part of this extraction and just assert that Dir.pwd + "config/secrets.default.yml" exists.  If we want to prevent core from using it, it can check if config/application.rb exists and blow up.

Reply:
I'm really confused - ENGINE_ROOT _is_ the path to the plugin and will always be available when running from a plugin.

Reply:
```ruby
~/dev/manageiq-providers-vmware $ be rails c
irb(main):001> ENGINE_ROOT
=> "/Users/jfrey/dev/manageiq-providers-vmware"
irb(main):002> Rails.root
=> #<Pathname:/Users/jfrey/dev/manageiq>
irb(main):003>
```

```ruby
~/dev/manageiq $ be rails c
irb(main):001> defined?(ENGINE_ROOT)
=> nil
irb(main):004> Rails.root
=> #<Pathname:/Users/jfrey/dev/manageiq>
```